dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -133,7 +132,7 @@ public GitHubSourceCodeRepo(String githubTokenUsername, String githubTokenConten
         OkHttpGitHubConnector okHttp3Connector = new OkHttpGitHubConnector(build, GITHUB_MAX_CACHE_AGE_SECONDS);
         try {
             this.github = new GitHubBuilder().withOAuthToken(githubTokenContent, githubTokenUsername)
-                    .withRateLimitHandler(new FailRateLimitHandler(githubTokenUsername))
+                    .withRateLimitChecker(new LiteralValue(SLEEP_AT_RATE_LIMIT_OR_BELOW))
```

Comment:
Do you know if this sleeps once or as long as it takes? IOW, while waiting, other requests have happened, and after sleeping, the rate limit is still exceeded, does it wait again?

Reply:
Was looking at the docs: https://github-api.kohsuke.org/apidocs/org/kohsuke/github/RateLimitChecker.html
Does this line sleep when the remaining number of available requests is less than or equal to `SLEEP_AT_RATE_LIMIT_OR_BELOW`?

Reply:
> How do you envision the disabling happening? Via a config setting? Asking because does the FailRateLimitHandler need to be preserved?

Thinking we key off of an environment variable, more details in https://github.com/dockstore/dockstore/pull/5348

Reply:
> while waiting, other requests have happened, and after sleeping, the rate limit is still exceeded, does it wait again?

Yes, if somehow some other job beats this to the punch, when the test wakes up, it would try again and then wait again.



> Does this line sleep when the remaining number of available requests is less than or equal to

Yes